app/controllers/participants/validations_controller.rb

```diff
@@ -0,0 +1,237 @@
+    end
+
+    def find_your_trn; end
+
+    def get_a_trn; end
+
+    def have_you_changed_your_name
+      if request.put?
+        if step_valid?
+          choice = @form.have_you_changed_your_name_choice
+          if choice == "yes"
+            redirect_to_step :confirm_updated_record
+          else
+            redirect_to_step :tell_us_your_details
+          end
+        end
+      end
+    end
+
+    def confirm_updated_record
+      if request.put?
+        if step_valid?
+          choice = @form.updated_record_choice
+          if choice == "yes"
+            redirect_to_step :tell_us_your_details
+          elsif choice == "no"
+            redirect_to_step :name_not_updated
+          else
+            redirect_to_step :check_with_tra
+          end
+        end
+      end
+    end
+
+    def name_not_updated
+      if request.put?
+        if step_valid?
+          choice = @form.name_not_updated_choice
+          if choice == "register_previous_name"
+            redirect_to_step :tell_us_your_details
+          else
+            redirect_to_step :change_your_details_with_tra
+          end
+        end
+      end
+    end
+
+    def change_your_details_with_tra; end
+
+    def check_with_tra; end
+
+    def tell_us_your_details
+      if request.put?
+        if step_valid?
+          redirect_to_step :confirm_details
+        end
+      end
+    end
+
+    def confirm_details
+      if request.put?
+        if step_valid?
+          validate_participant_details_and_redirect
+        end
+      end
+    end
+
+    def cannot_find_details
+      if request.put?
+        if step_valid?
+          store_validation_data!
+          reset_form_data
+          redirect_to_step :complete
+        end
+      end
+    end
+
+    def complete
+      @school = participant.school
+      if participant.ecf_participant_eligibility.present? && participant.ecf_participant_eligibility.eligible_status?
+        # TRN has been validated, qts, no flags, not done before, no previous induction
+        render_completed_page
+      else
+        # all other cases
+        render_manual_check_page
+      end
+    end
+
+    def reset
+      reset_form_data
+      redirect_to participants_validation_start_path
+    end
+
+  private
+
+    def render_completed_page
+      if participant.school_cohort.full_induction_programme?
+        @partnership = @school.partnerships.active.find_by(cohort: participant.school_cohort.cohort)
+        render "complete_fip"
+      else
+        render "complete_cip"
+      end
+    end
+
+    def render_manual_check_page
+      if participant.school_cohort.full_induction_programme?
+        @partnership = @school.partnerships.active.find_by(cohort: participant.school_cohort.cohort)
+        render "manual_details_check_fip"
+      else
+        render "manual_details_check_cip"
+      end
+    end
+
+    def check_not_already_completed
+      redirect_to_step :complete if complete?
+    end
+
+    def complete?
```

Comment:
What is complete exactly? `flow_complete`?

Reply:
Yes spot on. Once a participant has completed the flow they are directed straight to the "complete" step on any subsequent visits.